src/main/java/Expense.java

```diff
@@ -1,17 +1,91 @@
 package src.main.java;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
 public class Expense {
     private int expenseId;
     private String description;
     private double amount;
     private User paidBy;
+    private Group group;
+    private List<User> benefactors;
+    private double individualShare;
 
-    public Expense(int expenseId, String description, double amount, User paidBy) {
+    public Expense(int expenseId, String description, double amount, User paidBy, Group group) {
         this.expenseId = expenseId;
         this.description = description;
         this.amount = amount;
         this.paidBy = paidBy;
+        this.group = group;
+        this.benefactors = new ArrayList<>();
+    }
+
+    // Getters and Setters
+    public int getExpenseId() {
+        return expenseId;
+    }
+
+    public void setExpenseId(int expenseId) {
+        this.expenseId = expenseId;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    public double getAmount() {
+        return amount;
+    }
+
+    public void setAmount(double amount) {
+        this.amount = amount;
+    }
+
+    public User getPaidBy() {
+        return paidBy;
+    }
+
+    public void setPaidBy(User paidBy) {
+        this.paidBy = paidBy;
     }
 
-    // Getters for attributes
+    public Group getGroup() {
+        return group;
+    }
+
+    public void setGroup(Group group) {
+        this.group = group;
+    }
+
+    public List<User> getBenefactors() {
+        return benefactors;
+    }
+
+    public void addBenefactor(User user) {
+        benefactors.add(user);
+    }
+
+    public double getIndividualShare() {
+        return individualShare = calculateIndividualShare();
```

Comment:
this looks like a redundant assignment,

doesn't the `calculateIndividualShare`  act as a getter function for individual share? please callout the logic this code is structured this way.

am i missing some functionality that you plan to implement in future?

Reply:
You are correct, Jainam. This is a redundant assignment.

I have resolved this in the new commit. Please check it out and let me know if still any mistakes are there. 
Thank you!